### PR TITLE
feat(FEC-11284): dual screen core changes

### PR DESCRIPTION
**Pull request by RoyBregman**
### Description of the Changes

As part of the dual screen plugin development there are several changes that needed to be done in the ui:

- Fix dynamic icon existence check to support complete removal and adding of an icon again and again
- Icon now supports other sizes than 36x36px
- Move playlist and unmute indication to interactive area
- Make interactive area grow and shrink dynamically according to top / bottom
- Make top and bottom bars slide in-out
- Add blur on interactive area when overlay is active
- Interactive changes its margin according to wether the ui is open
- Bottom bar has a new state and class added to identify it was broken to two lines
- Bottom bar padding / margins were fixed with Dana (reduced size from 68px to 60px)
- Seek bar hover grow is now both up and down so it remains centred vertically
- Added new PlayerArea "VideoContainer" inside the video area to support side by side mode in dual screen

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/icon/icon.js [hunk 1]
```diff
@@ -102,11 +104,13 @@ class Icon extends Component {
...
    *
    * @param {string} path - svg path
+   * @param {?number} width - svg width
+   * @param {?number} height - svg height
    * @returns {string} - encoded svg url
    * @memberof Icon
    */
-  getSVGUrl = (path: string): string => {
-    const svg = `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 1024 1024" width="36" height="36">${path}</svg>`;
+  getSVGUrl = (path: string, width: ?number = 36, height: ?number = 36): string => {
```
Comment (dan-ziv) [inline src/components/icon/icon.js:112]: should be
`width?: number = 36, height?: number = 36`

**Comment by OrenMe** [comment]
@RoyBregman please only use FEC in ticket titles. @yairans FYI. Cross dependencies should be defined in FEC ticket and tested in core separately for regressions.

File: src/components/icon/icon.js (hunk 1 above)
Comment (RoyBregman) [reply src/components/icon/icon.js]: should be w/o "?" Actually . Fixed

**Comment by RoyBregman** [comment]
> @RoyBregman please only use FEC in ticket titles. @yairans FYI. Cross dependencies should be defined in FEC ticket and tested in core separately for regressions.

Done - updated PR title

File: src/components/seekbar/_seekbar.scss [hunk 2]
```diff
@@ -1,8 +1,8 @@
 @import '~styles/variables';
 
 .player .seek-bar {
-  padding: 12px 0 16px 0;
-  margin: -2px 0;
+  padding: 12px 0 12px 0;
+  //margin: -2px 0;
```
Comment (yairans) [inline src/components/seekbar/_seekbar.scss:5]: remove

File: src/components/interactive-area/_interactive-area.scss [hunk 3]
```diff
@@ -1,8 +1,42 @@
...
+  }
+}
+
+.player.casting,
+.player.metadata-loaded.hover,
+.player.state-paused,
+.player.state-idle,
+.player.ad-break,
+.player.menu-active {
+  &:not(.overlay-active) {
```
Comment (yairans) [inline src/components/interactive-area/_interactive-area.scss:37]: can we do it without duplicate all these? try to add the size-sm rule here

File: src/components/bottom-bar/_bottom-bar.scss [hunk 4]
```diff
@@ -13,6 +15,8 @@
   left: 0;
   z-index: 1;
   pointer-events: auto;
+  position: relative;
+  padding: 0 $gui-gutter 4px;
```
Comment (yairans) [inline src/components/bottom-bar/_bottom-bar.scss:19]: position and padding should be added to the top bar as well?

File: src/components/bottom-bar/_bottom-bar.scss [hunk 5]
```diff
@@ -66,27 +70,26 @@
...
+        max-height: 92px;
+      }
+    }
+  }
 }
 
 .player.size-sm {
   .bottom-bar {
-    padding: 6px 8px;
+    padding: 0 $gui-small-gutter 4px;
```
Comment (yairans) [inline src/components/bottom-bar/_bottom-bar.scss:92]: should be on left and right controls like in the top bar?

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/components/bottom-bar/_bottom-bar.scss (hunk 5 above)
Comment (RoyBregman) [reply src/components/bottom-bar/_bottom-bar.scss:92]: You are right it is different. But thats the way it was and I didn't want to change the css completely

**Comment by OrenMe** [comment]
If we change the UI vars in a way that requires updating the CSS guide than please update it

File: src/components/playlist-countdown/_playlist-countdown.scss [hunk 6]
```diff
@@ -12,8 +12,6 @@
     position: absolute;
     right: 0;
     bottom: 0;
-    margin: 16px;
-    transform: translateY(0px);
     transition: transform ease-in 100ms, opacity 400ms, bottom 300ms;
```
Comment (yairans) [inline src/components/playlist-countdown/_playlist-countdown.scss:15]:
```suggestion
    transition: opacity 400ms, bottom 300ms;
```

File: src/components/bottom-bar/_bottom-bar.scss [hunk 7]
```diff
@@ -1,18 +1,20 @@
...
+  transition: #{$hover-animation-time}ms max-height ease-in-out, #{$hover-animation-time}ms visibility ease-in-out;
   width: 100%;
   margin-top: auto;
-  -ms-align-self: flex-end;
-  align-self: flex-end;
   bottom: 0;
   left: 0;
   z-index: 1;
   pointer-events: auto;
+  position: relative;
```
Comment (yairans) [inline src/components/bottom-bar/_bottom-bar.scss:16]: missing in top bar?

File: src/components/top-bar/_top-bar.scss [hunk 8]
```diff
@@ -1,18 +1,28 @@
...
 .player .top-bar {
   background: linear-gradient(0deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.6) 100%);
   color: #fff;
-  opacity: 0;
   visibility: hidden;
-  height: 0;
-  transition: 100ms opacity;
+  height: auto;
+  max-height: 0;
+  transform: translateY(#{-$top-bar-max-height}px);
```
Comment (yairans) [inline src/components/top-bar/_top-bar.scss:9]: missing in bottom bar?

File: src/components/bottom-bar/_bottom-bar.scss (hunk 7 above)
Comment (yairans) [inline src/components/bottom-bar/_bottom-bar.scss:16]: missing in top bar?

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/components/bottom-bar/_bottom-bar.scss (hunk 7 above)
Comment (RoyBregman) [reply src/components/bottom-bar/_bottom-bar.scss:16]: yes, added now

File: src/components/top-bar/_top-bar.scss (hunk 8 above)
Comment (RoyBregman) [reply src/components/top-bar/_top-bar.scss:9]: No, Bottom is different in this case

Comment (RoyBregman) [reply src/components/top-bar/_top-bar.scss:9]: Actually, I removed the max-height in the top-bar. Topbar animates the translate and bottombar animates the max-height

File: src/components/interactive-area/_interactive-area.scss (hunk 3 above)
Comment (RoyBregman) [reply src/components/interactive-area/_interactive-area.scss]: done

File: src/components/playlist-countdown/_playlist-countdown.scss (hunk 6 above)
Comment (RoyBregman) [reply src/components/playlist-countdown/_playlist-countdown.scss:15]: done

File: src/components/seekbar/_seekbar.scss (hunk 2 above)
Comment (RoyBregman) [reply src/components/seekbar/_seekbar.scss:5]: done

File: src/components/video-player/video-player.js [hunk 9]
```diff
@@ -112,7 +113,11 @@ class VideoPlayer extends Component {
   render(): React$Element<any> {
     const {videoStyles, targetId} = this.props;
 
-    return <div id={`${targetId}-video`} className={style.videoPlayer} style={videoStyles} ref={this._setRef} />;
+    return (
+      <div id={`${targetId}-video`} className={style.videoPlayer} style={videoStyles} ref={this._setRef}>
+        <PlayerArea name={'VideoContainer'} />
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/components/video-player/video-player.js:118]: We agreed it is ok

File: src/ui-presets/live.js [hunk 10]
```diff
@@ -81,6 +80,7 @@ class LiveUI extends Component {
                 <Fragment>
                   <TopBar rightControls={<Share />} />
                   <InteractiveArea>
+                    <UnmuteIndication />
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/ui-presets/live.js:83]: done

File: src/ui-presets/playback.js [hunk 11]
```diff
@@ -87,7 +85,9 @@ class PlaybackUI extends Component {
                 <Fragment>
                   <TopBar rightControls={<Share />} />
                   <InteractiveArea>
+                    <UnmuteIndication />
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/ui-presets/playback.js:88]: done

File: src/components/bottom-bar/_bottom-bar.scss (hunk 4 above)
Comment (RoyBregman) [reply src/components/bottom-bar/_bottom-bar.scss]: done

File: src/components/icon/icon.js (hunk 1 above)
Comment (RoyBregman) [reply src/components/icon/icon.js]: done

**Review by yairans — APPROVED** [review]

**Pull request merged**